HLTrigger/btau/plugins/HLTSumJetTag.cc

```diff
@@ -0,0 +1,137 @@
+
+  typedef vector<T> TCollection;
+  typedef Ref<TCollection> TRef;
+
+  edm::Handle<TCollection> h_Jets;
+  event.getByToken(m_JetsToken, h_Jets);
+  if (saveTags())
+    filterproduct.addCollectionTag(m_Jets);
+
+  edm::Handle<JetTagCollection> h_JetTags;
+  event.getByToken(m_JetTagsToken, h_JetTags);
+
+  // check if the product this one depends on is available
+  auto const& handle = h_JetTags;
+  auto const& dependent = handle->keyProduct();
+  if (not dependent.isNull() and not dependent.hasCache()) {
+    // only an empty AssociationVector can have a invalid dependent collection
+    edm::Provenance const& dependent_provenance = event.getProvenance(dependent.id());
+    if (dependent_provenance.branchDescription().dropped())
+      // FIXME the error message should be made prettier
+      throw edm::Exception(edm::errors::ProductNotFound)
+          << "Product " << handle.provenance()->branchName() << " requires product "
+          << dependent_provenance.branchName() << ", which has been dropped";
+  }
+
+  //// Loop on jet tags and store values, sorting them in decreasing order
+  std::vector<float> jetTag_values;
+  for (auto const& jet : *h_JetTags)
+    jetTag_values.push_back(jet.second);
+  std::sort(jetTag_values.begin(), jetTag_values.end(), std::greater<>());
+
+  //// apply max-jet requirement --> store only tags up to m_MaxJetToSum
+  std::vector<float> jetTag_values_selected;
+  unsigned int nJet = 0;
+  for (auto const& jetTag : jetTag_values) {
+    if (nJet >= m_MaxJetToSum)
+      break;
+    ++nJet;
+    jetTag_values_selected.push_back(jetTag);
+  }
+
+  /// produce jet collection
+  std::vector<TRef> jetRefCollection;
+  TRef jetRef;
+  float sumJetTag = 0;
+  nJet = 0;
+  for (auto const& jet : *h_JetTags) {
+    jetRef = TRef(h_Jets, jet.first.key());
+    LogTrace("") << "Jet " << nJet << " : Et = " << jet.first->et() << " , tag value = " << jet.second;
+    nJet++;
+    if (std::find(jetTag_values_selected.begin(), jetTag_values_selected.end(), jet.second) !=
+        jetTag_values_selected.end()) {  // if found
+      jetRefCollection.push_back(jetRef);
+      sumJetTag += jet.second;
+    }
+  }
+
+  if (m_UseMeanValue)
+    sumJetTag /= jetTag_values_selected.size();
+
+  // Accept value of the filter
+  bool accept = true;
+  if (jetRefCollection.size() < m_MinJetToSum)
+    accept = false;
+  else {
+    if (sumJetTag >= m_MinTag and sumJetTag <= m_MaxTag) {
+      accept = true;
+      for (auto const& jetRef : jetRefCollection) {
+        filterproduct.addObject(m_TriggerType, jetRef);
+      }
+    } else
+      accept = false;
+  }
+
+  edm::LogInfo("") << " trigger accept ? = " << accept << " nTag/nJet = " << jetRefCollection.size() << "/" << nJet
+                   << std::endl;
+  return accept;
+}
+
+typedef HLTSumJetTag<reco::CaloJet> HLTSumCaloJetTag;
+typedef HLTSumJetTag<reco::PFJet> HLTSumPFJetTag;
+DEFINE_FWK_MODULE(HLTSumCaloJetTag);
+DEFINE_FWK_MODULE(HLTSumPFJetTag);
```

Comment:
Hi @missirol thanks for the comments! I'll implement all your suggestions. About merging the .cc with the header into a single file, I haven't done it since there are other template plugins in the btau area that have two distinct files (HLTJetPairDzMatchFilter, HLTJetTag..) .. therefore I was following that style. Anyway I'll update the SumJetTag filters accordingly.